### ISSUE TITLE
 Is it possible to resolve the schema, but not expand the paths when doing a `collect_schema`?

### ISSUE THREAD

**Issue by kszlim** [pola-rs/polars#17855]
### Description

Path expansion for large datasets is very expensive, but I imagine schema resolution itself might be very cheap (relative to path expansion). Is it possible for schema resolution to avoid path expansion? It seems to take pretty long for me on a 10k file dataset, (i'm guessing it's due to listing all the paths).

**Comment by stinodego** [comment]
Ideally we shouldn't do path expansion if the reader is provided a schema, e.g. through a `schema` argument. But I'm not sure where we're at with the implementation. Tagging @nameexhaustion for this one.

**Comment by nameexhaustion** [comment]
> Is it possible for schema resolution to avoid path expansion?

We can't - we need to get the schema from the file, which needs the resolved file path. 

> Ideally we shouldn't do path expansion for collect_schema if the reader is provided a schema

This seems reasonable to me, although I don't think we're currently in a good position to do this - e.g. `scan_parquet` currently doesn't accept any schema argument, and I'm not sure if the `schema` argument of the other functions behave in a consistent manner (e.g. particularly `scan_csv`)

**Comment by ritchie46** [comment]
No, and we shouldn't do this. This is all a band aid to a proper solution that is that we need the schema after path expansion. We are working towards that.

**Comment by ritchie46** [comment]
The point is that we shouldn't collect the schema internally until we are running the query.
